flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/validation/CalcValidationTest.scala

```diff
@@ -46,4 +47,88 @@ class CalcValidationTest extends TableTestBase {
     .groupBy('w)
     .select('w.end.rowtime, 'int.count as 'int) // no rowtime on non-window reference
   }
+
+  @Test
+  def testAddColumns(): Unit = {
+    val util = streamTestUtil()
+    val tab = util.addTable[(Int, Long, String)]("Table3",'a, 'b, 'c)
+
+    // Test aggregates
+    try {
+      tab.addColumns('a.sum)
+      fail("TableException expected")
+    } catch {
+      case _: TableException => //ignore
+    }
+
+    // Test replace the existing column, but do not with the alias.
+    try {
+      tab.addColumns(true, concat('c, "Sunny"))
+      fail("TableException expected")
+    } catch {
+      case _: TableException => //ignore
+    }
+
+  }
+
+  @Test
+  def testRenameColumns(): Unit = {
+    val util = streamTestUtil()
+    val tab = util.addTable[(Int, Long, String)]("Table3",'a, 'b, 'c)
+
+    // Test aggregates
+    try {
+      tab.renameColumns('a.sum)
+      fail("TableException expected")
+    } catch {
+      case _: TableException => //ignore
+    }
+
+    // Test without alias case.
+    try {
+      tab.renameColumns('a)
+      fail("TableException expected")
+    } catch {
+      case _: TableException => //ignore
+    }
+
+    // Test for fields that do not exist
+    try {
+      tab.renameColumns('e)
```

Comment:
Add a rename test such as the following to make sure that only existing fields can be renamed:
tab.renameColumns('a + 1 as 'e)

Reply:
I think line 97 has test the field which not exist case. but i should add the alias. what do you think?